### ISSUE TITLE
Redis Master does not come up

### ISSUE THREAD

**Issue by adurai81** [amaizfinance/redis-operator#8]
I ran through the steps, but there are no redis instances available.

```
kubectl create namespace redis-operator
kubectl apply -Rf deploy
kubectl -n redis-operator get deployment
kubectl create -f example/k8s_v1alpha1_redis_cr.yaml
kubectl get redis example

NAME      MASTER   REPLICAS   DESIRED   AGE
example                       3         10h
```

Additional Info
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:49:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.7-3+1a4d39a9763d23", GitCommit:"1a4d39a9763d23548d16e262d6dbe0fb2198a8ce", GitTreeState:"clean", BuildDate:"2019-03-27T01:47:38Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

Can you pls take a look ?

Let me know if you need any logs/additional detail.

**Comment by nrvnrvn** [comment]
Hi! 

Is there anything in logs of the operator?
Has it created all services, statefulset, et al.?

**Comment by step76** [comment]
We had the same problem.

We noticed that the probem happens when the instances have some data to load at start, while loading the dataset a redis instance is live (respond PONG to PING), ma does not accept commands.

We solved adding the initialDelaySeconds and setting it to 120 seconds.

https://github.com/mmbsoftware-it/redis-operator/commit/3a63f7fdce9af396e382d0194b6bc90cca062e0b

**Comment by nrvnrvn** [comment]
thanks for the explanation @step76!

What size is the dataset? I wonder if I could reproduce this behaviour.

Also, your fork looks quite interesting! I examined it a bit and have a small question: why did you decide to add an init container for disabling transparent huge pages instead of doing the same [via a DaemonSet](https://medium.com/@allanlei/disabling-transparent-huge-pages-in-kubernetes-9e8f60141fb1) with properly set up [Pod Priorities](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#pod-priority)?

I mean there are multiple ways to deal with kernel settings. I am curious why you chose the `initContainers` path.

**Comment by step76** [comment]
Our dataset is about 1GB and on our GKE cluster takes 30/40 seconds to load on startup.

I don't like the DaemonSet approach because it tends to introduce too many dependencies. For example, in the article you mentioned I don't like the approach of assigning a label to the node.

initContainers guarantees the order of execution, moreover the POD dies before the main one starts and I have everything in a single YAML file.

But now I want to investigate better the Pod Priorities that I didn't know about.

If you want I can submit PR.

**Comment by nrvnrvn** [comment]
yep, a PR would be great.

I would be happy to know your findings about Pod Priorities as well. We are currently relying on them for ordered scheduling of pods.

**Comment by nrvnrvn** [comment]
fixed in #9
